src/OpenTelemetry.Exporter.Geneva/MsgPackExporter/MsgPackLogExporter.cs

```diff
@@ -288,6 +285,11 @@ internal int SerializeLogRecord(LogRecord logRecord)
                     cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, entry.Key);
                     cursor = MessagePackSerializer.Serialize(buffer, cursor, entry.Value);
                     cntFields += 1;
+
+                    if (entry.Key == "name")
```

Comment:
I don't love that we are double-checking the key here. We could cache that result. Consider something like...

```csharp
        bool hasEnvProperties = false;
        bool bodyPopulated = false;
        bool namePopulated = false;
        for (int i = 0; i < listKvp?.Count; i++)
        {
            var entry = listKvp[i];

            // Iteration #1 - Get those fields which become dedicated columns
            // i.e all Part B fields and opt-in Part C fields.
            if (entry.Key == "{OriginalFormat}")
            {
                cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "body");
                cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, logRecord.FormattedMessage ?? Convert.ToString(entry.Value, CultureInfo.InvariantCulture));
                cntFields += 1;
                bodyPopulated = true;
                continue;
            }

            var isNameEntry = string.Equals(entry.Key, "name", StringComparison.OrdinalIgnoreCase);

            if (isNameEntry || this.m_customFields == null || this.m_customFields.ContainsKey(entry.Key))
            {
                // TODO: the above null check can be optimized and avoided inside foreach.
                if (entry.Value != null)
                {
                    // null is not supported.
                    cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, entry.Key);
                    cursor = MessagePackSerializer.Serialize(buffer, cursor, entry.Value);
                    cntFields += 1;

                    if (isNameEntry)
                    {
                        namePopulated = true;
                    }
                }
            }
            else
            {
                hasEnvProperties = true;
                continue;
            }
        }

        if (!namePopulated)
        {
            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, "name");
            cursor = MessagePackSerializer.SerializeUnicodeString(buffer, cursor, categoryName);
            cntFields += 1;
        }
```

Reply:
Good idea, but ended up not having to double-check at all to avoid breaking back compat when customFields is set but doesn't include "name"